### PR TITLE
Mokito

### PR DESCRIPTION
**Pull request by RobertRodeapps**
Integrations tests

### CODE REVIEW

File: src/main/java/com/problem1/myproject/controller/UserRestController.java [hunk 1]
```diff
@@ -122,15 +123,15 @@ public UserUpdateDTO deleteUser(@PathVariable long userId){      ///TODO ASK IF
...
+/*    @PutMapping("/buy/{userId}")  ///TODO MAYBE ADD QUANTITY LATER
     @Transactional
     public List<Coin> buyCoin(@PathVariable long userId, @RequestBody Coin theCoin){
 
 
         return  this.userService.buyCoin(userId,theCoin);
 
 
-    }   ///TODO it doesnt work, i think its harder than i thought
+    }   ///TODO it doesnt work, i think its harder than i thought*/
```
Comment (DragosMoldovan) [inline src/main/java/com/problem1/myproject/controller/UserRestController.java:134]: @RobertRodeapps this is related to the M-M relation? What exactly doesn't work?

File: src/main/java/com/problem1/myproject/service/implementation/UserService.java [hunk 2]
```diff
@@ -24,12 +23,14 @@
 public class UserService implements IUserService, UserDetailsService {
 
     private UserRepository userRepo;
-    @Autowired
+
+    //@Autowired
```
Comment (DragosMoldovan) [inline src/main/java/com/problem1/myproject/service/implementation/UserService.java]: @RobertRodeapps you can remove this comment

File: src/main/java/com/problem1/myproject/service/implementation/UserService.java [hunk 3]
```diff
@@ -24,12 +23,14 @@
...
     private PasswordEncoder passwordEncoder;
 
     @Autowired
-    public UserService(UserRepository userRepo) {
+    public UserService(UserRepository userRepo, PasswordEncoder passwordEncoder ) {
```
Comment (DragosMoldovan) [inline src/main/java/com/problem1/myproject/service/implementation/UserService.java]: @RobertRodeapps Please replace this constructor with a AllArgsConstructor annotation

File: src/test/java/com/problem1/myproject/MyprojectApplicationTests.java [hunk 4]
```diff
@@ -2,12 +2,18 @@
...
 
 @SpringBootTest
 class MyprojectApplicationTests {
 
 	@Test
 	void contextLoads() {
 	}
-
+	@Bean
+	PasswordEncoder passwordEncoder(){
```
Comment (DragosMoldovan) [inline src/test/java/com/problem1/myproject/MyprojectApplicationTests.java]: @RobertRodeapps why did you define this bean in a test class?

File: src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        Coin firstCoin = createCoin(2,"xrp",12.1,3213.22);
+        Coin secondCoin = createCoin(3,"egld",123.1,3211233.22);
+
+        given(coinRepo.findAll()).willReturn(asList(firstCoin, secondCoin));
+
+        //when
+        List<Coin> receivedCoins = coinService.findAll();
+        //then
+        assertThat(receivedCoins).asList().containsExactly(firstCoin,secondCoin);
+        verify(coinRepo).findAll();
```
Comment (DragosMoldovan) [inline src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java]: @RobertRodeapps please also specify the number of times that this method was called

File: src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+    }
+
+    @Test
+    void cannotFindById() {
+
+        given(coinRepo.findById(1L)).willReturn(Optional.empty());
+/*
+        when(coinService.findById(1L)).thenThrow(new ObjectNotFoundException("Did not find Coin id - 1"));
```
Comment (DragosMoldovan) [inline src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java]: @RobertRodeapps please delete useless comments

File: src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        /*given(otherServiceMock.bar()).willThrow(new MyException());
+
+        when(() -> myService.foo());
+
+        then(caughtException()).isInstanceOf(MyException.class);*/
+
+    }
+
+    @Test
+    void canAddCoin() {
```
Comment (DragosMoldovan) [inline src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java:69]: @RobertRodeapps without a mocking of the coinRepo, I think that the following tests will throw a NPE

File: src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        Coin coin = createCoin(1,"ada",12.2,100.2);
+
+        //when
+        coinService.save(coin);
+
+        //then
+        ArgumentCaptor<Coin> coinArgumentCaptor =
```
Comment (DragosMoldovan) [inline src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java]: @RobertRodeapps, please do not use ArgumentCaptor. You should save the service response in a variable and if you want to use generic types for mock methods or verify methods, you can use the any() method from mockito (any(), any(Coin.class))

**Review by DragosMoldovan — CHANGES_REQUESTED** [review]
@RobertRodeapps please remove all unused comments and format your code using the given AndroidStyle XML

File: src/main/java/com/problem1/myproject/controller/UserRestController.java (hunk 1 above)
Comment (RobertRodeapps) [reply src/main/java/com/problem1/myproject/controller/UserRestController.java:134]: pot sa salvez coin ul in portofoliu(atribut a unui User), dar ar trebui salvat si faptul ca Coin -ul e acum detinut de un nou  User si ar trebui sa l adaug pe User ul ce a cumparat coin ul respectiv in lista de holders(atribut a unui Coin)
si nu am adresa in memorie a Coin ului
parca primeam ceva eroare de la relatia M-M

File: src/test/java/com/problem1/myproject/MyprojectApplicationTests.java (hunk 4 above)
Comment (RobertRodeapps) [reply src/test/java/com/problem1/myproject/MyprojectApplicationTests.java]: I defined it when i have not understood very well what @Mock Does and i forgot about it, i will erase it

File: src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java (hunk 8 above)
Comment (RobertRodeapps) [reply src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java]: well the service will not return anything, the save function is a void function , should i make save functiion return the saved object ?

Comment (DragosMoldovan) [reply src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java]: yes

**Pull request closed without merging**